jib-maven-plugin/src/integration-test/java/com/google/cloud/tools/jib/maven/BuildImageMojoIntegrationTest.java

```diff
@@ -686,6 +695,85 @@ public void testExecute_springBootPackaged()
+
+    Verifier verifier = new Verifier(simpleTestProject.getProjectRoot().toString());
+    verifier.setSystemProperty("_TARGET_IMAGE", targetImage);
+
+    verifier.setSystemProperty("jib.to.auth.username", "testuser");
+    verifier.setSystemProperty("jib.to.auth.password", "testpassword");
+    verifier.setSystemProperty("sendCredentialsOverHttp", "true");
+    verifier.setSystemProperty("jib.allowInsecureRegistries", "true");
+
+    verifier.setAutoclean(false);
+    verifier.addCliOption("-X");
+    verifier.addCliOption("--file=pom-multiplatform-build.xml");
+    verifier.executeGoals(Arrays.asList("clean", "compile", "jib:build"));
+    verifier.verifyErrorFreeLog();
+
+    FailoverHttpClient httpClient = new FailoverHttpClient(true, true, ignored -> {});
+    RegistryClient registryClient =
+        RegistryClient.factory(EventHandlers.NONE, "localhost:5000", "multiplatform", httpClient)
+            .setCredential(Credential.from("testuser", "testpassword"))
+            .newRegistryClient();
+    registryClient.configureBasicAuth();
+
+    // manifest list by tag ":latest"
+    ManifestTemplate manifestListTemplate = registryClient.pullManifest("latest").getManifest();
+    MatcherAssert.assertThat(
+        manifestListTemplate, CoreMatchers.instanceOf(V22ManifestListTemplate.class));
+    V22ManifestListTemplate manifestList = (V22ManifestListTemplate) manifestListTemplate;
+    Assert.assertEquals(
+        Arrays.asList("sha256:fee2655e19e5138150606c99cfc16fcbf502d72b0f3b9ccf3a8f4509c47e46d9"),
+        manifestList.getDigestsForPlatform("arm64", "linux"));
+    Assert.assertEquals(
+        Arrays.asList("sha256:f3f4a91c68bcafea351280085d17e25fa598f5644c8b5e31e6133eddfc35e7ff"),
+        manifestList.getDigestsForPlatform("amd64", "linux"));
+
+    // manifest list by tag ":another"
+    ManifestTemplate manifestListTemplate2 = registryClient.pullManifest("another").getManifest();
+    MatcherAssert.assertThat(
+        manifestListTemplate2, CoreMatchers.instanceOf(V22ManifestListTemplate.class));
+    V22ManifestListTemplate manifestList2 = (V22ManifestListTemplate) manifestListTemplate2;
+    Assert.assertEquals(
+        Arrays.asList("sha256:fee2655e19e5138150606c99cfc16fcbf502d72b0f3b9ccf3a8f4509c47e46d9"),
+        manifestList2.getDigestsForPlatform("arm64", "linux"));
+    Assert.assertEquals(
+        Arrays.asList("sha256:f3f4a91c68bcafea351280085d17e25fa598f5644c8b5e31e6133eddfc35e7ff"),
+        manifestList2.getDigestsForPlatform("amd64", "linux"));
+
+    // arm64/linux manifest
+    ManifestAndDigest<ManifestTemplate> manifestAndDigest1 =
+        registryClient.pullManifest(
+            "sha256:fee2655e19e5138150606c99cfc16fcbf502d72b0f3b9ccf3a8f4509c47e46d9");
+    Assert.assertEquals(
+        "sha256:fee2655e19e5138150606c99cfc16fcbf502d72b0f3b9ccf3a8f4509c47e46d9",
+        manifestAndDigest1.getDigest().toString());
+    ManifestTemplate manifestTemplate1 = manifestAndDigest1.getManifest();
```

Comment:
Can we move line 755 to be below line 749? To make the assert blocks be separate from the initialization code.